modules/BookCollection.js

```diff
@@ -0,0 +1,45 @@
+import Book from './Book.js';
+
+export default class BookCollection {
+  constructor() {
+    this.books = JSON.parse(localStorage.getItem('books')) || [];
+  }
+
+  // Using classes instead of objects for add book function.
+  addBook(title, author) {
+    const book = new Book(title, author);
+
+    this.books.push(book);
+    this.saveToLocalStorage();
+  }
```

Comment:
👍 Your `addBook` function is successfully adding new books, and that's great! However, you are working with a single-page app, which means that adding a new book should not trigger a page refresh. Currently, after adding a new book, the page is refreshing, which is not in line with the single-page app requirement. To meet this requirement, please update your code and implement a solution that allows adding a new book without triggering a page refresh. This could involve using JavaScript to dynamically update the content on the page without reloading the entire page.

[Awesome-Books-with-ES6-Modules (2).webm](https://github.com/Omwomo/Awesome-books-with-ES6/assets/55840999/819f97e9-06b4-44fc-97bd-5124aab3e084)

Adhering to the single-page app concept will enhance the user experience and ensure the project meets the specified functionality. Keep up the good work, and I'm confident you'll handle this update effectively! 📚🚀